provisioner/clusterpy_test.go

```diff
@@ -514,6 +516,40 @@ func TestPerformDeletion(t *testing.T) {
 				Name: "ns-foo",
 				Kind: "Namespace",
 			},
+			clientReactors: func(state map[string]interface{}) []k8stesting.SimpleReactor {
+				return []k8stesting.SimpleReactor{
+					{
+						Verb:     "patch",
+						Resource: "namespaces",
+						Reaction: func(action k8stesting.Action) (handled bool, ret runtime.Object, err error) {
+							payload := map[string]interface{}{}
+							err = json.Unmarshal(action.(k8stesting.PatchAction).GetPatch(), &payload)
+							if err != nil {
+								return false, nil, err
+							}
+							for k, v := range payload {
+								state[k] = v
+							}
+							return true, nil, nil
+						},
+					},
+				}
+			},
+			clientErrors: func(state map[string]interface{}) []clientError {
+				if meta, ok := state["metadata"]; ok {
+					if annotations, ok := meta.(map[string]interface{})["annotations"]; ok {
+						if _, ok := annotations.(map[string]interface{})["zalando.org/delete-date"]; ok {
+							return nil
+						}
+					}
+				}
+				return []clientError{
+					{
+						matches: deleteAction("namespaces", "ns-foo"),
+						err:     fmt.Errorf("admission webhook \"namespace-admitter.teapot.zalan.do\" denied the request: annotation zalando.org/delete-date not set in manifest to allow resource deletion. See https://cloud.docs.zalando.net/howtos/delete-protection/"),
+					},
+				}
+			},
```

Comment:
How come these errors don't change the `expectDeleted` outcome of this test case?

Reply:
Since I converted it to a function, I now check if the status map has the delete annotations, then it returns a nil

so, the error isn't actually raised and thus the output does not change